srml/treasury/src/lib.rs

```diff
@@ -291,8 +293,12 @@ impl<T: Trait> OnDilution<BalanceOf<T>> for Module<T> {
 		// pre dilution and post-dilution.
 		if !minted.is_zero() && !portion.is_zero() {
 			let total_issuance = T::Currency::total_issuance();
-			let funding = (total_issuance - portion) / portion * minted;
-			<Pot<T>>::mutate(|x| *x += funding);
+			if let Some(funding) = total_issuance.checked_sub(&portion) {
+				let funding = funding / portion;
```

Comment:
If `funding < portion`, this will be zero, so it appears that we might not be able to make it progress.

Reply:
Yeah true. So if `2 * portion > total_issuance` then it will halt. And it is the goal that a large amount of the network is staked. I was thinking about another edge case where `portion` is much smaller than `total_issuance` and `minted` is large (say a network with low stake and high rewards), then there is a risk of overflow.

However, how is this implementation different from the original? When `(total_issuance - portion) / portion * minted` gets evaluated, would `(total_issuance - portion) / portion` evaluate to zero and therefor not mutate the `Pot`?

Wondering if there is a better solution with `Permill`?

Reply:
> However, how is this implementation different from the original?

This is not to say that the original is correct... just trying to make sure my "fix" didn't break anything by accident.

Reply:
practically speaking, this won't be an issue as dilution will be capped at around 10% a year and given out in hourly portions.